apps/ledger-live-mobile/src/screens/SyncOnboarding/SoftwareChecksStep.tsx

```diff
@@ -0,0 +1,402 @@
+import React, { useCallback, useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { BoxedIcon, Flex, InfiniteLoader, Text } from "@ledgerhq/native-ui";
+import {
+  CircledAlertMedium,
+  CircledCheckSolidMedium,
+} from "@ledgerhq/native-ui/assets/icons";
+import { FlexBoxProps } from "@ledgerhq/native-ui/components/Layout/Flex";
+import { Device } from "@ledgerhq/live-common/hw/actions/types";
+import { useGenuineCheck } from "@ledgerhq/live-common/hw/hooks/useGenuineCheck";
+import { useGetLatestAvailableFirmware } from "@ledgerhq/live-common/hw/hooks/useGetLatestAvailableFirmware";
+import { getDeviceModel } from "@ledgerhq/devices";
+import { useTheme } from "styled-components/native";
+
+import GenuineCheckDrawer from "./GenuineCheckDrawer";
+import FirmwareUpdateDrawer from "./FirmwareUpdateDrawer";
+import GenuineCheckCancelledDrawer from "./GenuineCheckCancelledDrawer";
+import UnlockDeviceDrawer from "./UnlockDeviceDrawer";
+import AllowManagerDrawer from "./AllowManagerDrawer";
+
+const softwareStepDelay = 2500;
+const lockedDeviceTimeoutMs = 1000;
+
+type CheckStatus = "inactive" | "active" | "completed" | "failed";
+
+type CurrentSoftwareChecksStep = "none" | "genuine-check" | "firmware-update";
+
+type GenuineCheckUiStepStatus = CheckStatus;
+type GenuineCheckStatus =
+  | "unchecked"
+  | "requested"
+  | "ongoing"
+  | "completed"
+  | "failed";
+type GenuineCheckUiDrawerStatus =
+  | "none"
+  | "requested"
+  | "allow-manager"
+  | "unlock-needed"
+  | "cancelled";
+
+type FirmwareUpdateUiStepStatus = CheckStatus;
+type FirmwareUpdateStatus = "unchecked" | "ongoing" | "completed" | "failed";
+type FirmwareUpdateUiDrawerStatus = "none" | "new-firmware-available";
+
+type CheckCardProps = FlexBoxProps & {
+  title: string;
+  index: number;
+  status: CheckStatus;
+};
+
+const CheckCard = ({ title, index, status, ...props }: CheckCardProps) => {
+  const { colors } = useTheme();
+
+  const getCheckIcon = useCallback(
```

Comment:
I don't think you need a useCallback for this as you are not using this function directly as a prop (just calling it and passing the result as a prop).
Btw you could skip using a function altogether as the params are coming directly from the props.